tools/mkmodel/mkmodel.c

```diff
@@ -552,13 +791,25 @@ int convert(const char *infn, const char *outfn)
 
     cgltf_load_buffers(&options, data, infn);
 
-    model64_t *model = model64_alloc();
+    model64_data_t *model = model64_alloc();
 
     if (data->meshes_count <= 0) {
         fprintf(stderr, "Error: input file contains no meshes\n");
         goto error;
     }
-
+    if (data->nodes_count <= 0) {
+        fprintf(stderr, "Error: input file contains no nodes\n");
+        goto error;
+    }
+    if (!data->scene) {
+        fprintf(stderr, "Error: input file has no default scene\n");
+        goto error;
+    }
+    if (data->scene->nodes_count <= 0) {
+        fprintf(stderr, "Error: input file has no root node in default scene\n");
+        goto error;
+    }
```

Comment:
Do we have to error out here? What if someone just wants to import some meshes and handle everything else (transform, animation etc.) themselves? Obviously skinned meshes will need nodes, but skins can only be defined on nodes anyway.

Reply:
I changed it not to error out.